### PR TITLE
Fix crashes in subject hierarchy and DICOM reading

### PR DESCRIPTION
**Pull request by lassoan**
This pull request contains fixes for two independent problems that caused Slicer crashes.

### CODE REVIEW

**Review by jcfr — APPROVED** [review]
Good catch :+1: 

Nitpick:

I suggest to move the indentation changes into a `STYLE` commit.

Using `https://github.com/Slicer/Slicer/pull/7250/commits/dcf4b7d144584be099d95885e3d9368f9db1dbdc?w=1` (not the use of `?w=1`) helped me confirm the change as smaller than anticipated.

**Comment by lassoan** [comment]
> I suggest to move the indentation changes into a `STYLE` commit.

The indentation had to be changed because the added `try {} catch {}` block. If I moved the indentation fix to a separate commit then my commit wouldn't pass style checks (if they were enabled for C++) and would add noise to the commit history.

It is great that you've found this `?w=1` option because that makes review easier without forcing whitespace changes into a separate commit.

**Comment by jcfr** [comment]
> The indentation had to be changed because the added `try {} catch {}` block

Makes sense. 

> re: `w=1`

And reading https://github.blog/changelog/2021-10-14-hiding-whitespace-is-now-remembered-for-each-pull-request/

reminded me it was now available as part of the GitHub UI

image

File: Libs/vtkITK/vtkITKArchetypeImageSeriesReader.cxx [hunk 1]
```diff
@@ -777,189 +790,210 @@ int vtkITKArchetypeImageSeriesReader::RequestInformation(
...
+            {
+            scalarType = VTK_FLOAT;
+            }
+          else if( max <= std::numeric_limits<double>::max() )
+            {
+            scalarType = VTK_DOUBLE;
+            }
+          else
+            {
+            throw std::string("Cannot find suitable scalar type for value range");
```
Comment (jcfr) [inline Libs/vtkITK/vtkITKArchetypeImageSeriesReader.cxx:939]:
```suggestion
            // While throwing `std::string` is not considered to be best-practice, it
            // is done here to be consistent with how GDCM throw exception.
            throw std::string("Cannot find suitable scalar type for value range");
```

**Pull request merged**